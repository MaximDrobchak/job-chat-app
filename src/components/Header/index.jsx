import React, { Component } from 'react';
import injectSheet from 'react-jss';
import PropTypes from 'prop-types';
import settingIcon from '../../assets/settings.svg';
import Button from '../../elements/Button';
const styles = {
	root: {
		display: 'flex',
		justifyContent: 'space-between',
		padding: {
			top: 20,
			bottom: 13,
			left: 20,
			right: 20,
		},
		'& h1': {
			fontFamily: 'Roboto',
			fontSize: 22,
			fontWeight: 500,
		},
	},
};

const user = {
	name: 'Terry Crews',
};
class Header extends Component {
	render() {
		const { classes } = this.props;
		return (
			<div className={classes.root}>
				<h1>{user.name}</h1>
				<Button>
					<img src={settingIcon} alt="setting" />
				</Button>
			</div>
		);
	}
}
Header.propTypes = {
	classes: PropTypes.object.isRequired,
};
export default injectSheet(styles)(Header);
