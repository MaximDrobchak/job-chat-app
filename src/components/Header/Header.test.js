import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './index.jsx';

describe('Header', () => {
	it('отрисовывает без ошибки', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Header />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	test('есть корректный снимок', () => {
		const component = renderer.create(<Header />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
