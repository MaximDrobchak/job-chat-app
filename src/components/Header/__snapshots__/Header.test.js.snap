// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header есть корректный снимок 1`] = `
<div
  className="Header-root-0-1-1"
>
  <h1>
    Terry Crews
  </h1>
  <span
    className="Button-root-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="setting"
      src="settings.svg"
    />
  </span>
</div>
`;
